### PR TITLE
[libc] Implement forward iterators for libc fixed_vector

### PR DESCRIPTION
**Pull request by jameshu15869**
- Implements forward iterators for `cpp::fixed_vector` to use in https://github.com/llvm/llvm-project/pull/92009

### CODE REVIEW

File: libc/test/src/__support/CPP/array_test.cpp [hunk 1]
```diff
@@ -28,6 +28,11 @@ TEST(LlvmLibcArrayTest, Basic) {
   ASSERT_EQ(*(++it), 1);
   ASSERT_EQ(*(++it), 0);
 
+  auto forward_it = a.begin();
```
Comment (jhuber6) [inline libc/test/src/__support/CPP/array_test.cpp:31]: This tests `cpp::array` and not `cpp::fixed_vector`.

Comment (jameshu15869) [reply libc/test/src/__support/CPP/array_test.cpp:31]: Ah, I totally made a mistake

File: libc/src/__support/fixedvector.h [hunk 2]
```diff
@@ -63,6 +63,10 @@ template <typename T, size_t CAPACITY> class FixedVector {
     return reverse_iterator{&store[item_count]};
   }
   LIBC_INLINE constexpr reverse_iterator rend() { return store.rend(); }
+
+  using iterator = typename cpp::array<T, CAPACITY>::iterator;
+  LIBC_INLINE constexpr iterator begin() { return store.begin(); }
+  LIBC_INLINE constexpr iterator end() { return iterator{&store[item_count]}; }
```
Comment (jhuber6) [inline libc/src/__support/fixedvector.h:69]: Does returning `store.end()` not work?
```suggestion
  LIBC_INLINE constexpr iterator end() { return store.end(); }
```

Comment (jhuber6) [reply libc/src/__support/fixedvector.h:69]: Ah, that makes sense. I'm wondering if something like
`return store.begin() + item_count` would work better.

Comment (jameshu15869) [reply libc/src/__support/fixedvector.h:69]: Would that account for the size of the objects themselves when doing the pointer arithmetic? i.e. if an int is 4 bytes and you have 4 elements would you want `end()` to be at `store.begin() + 4*4`?

Comment (jameshu15869) [reply libc/src/__support/fixedvector.h:69]: Gotcha, I was waiting on rebuilding LLVM this morning but I can try it when I get home later today

Comment (lntue) [reply libc/src/__support/fixedvector.h:69]: I actually prefer `&store[item_count]` and let the compiler do the pointer arithmetic for you.

Comment (jhuber6) [reply libc/src/__support/fixedvector.h:69]: Just using the iterator directly makes sure it's the correct iterator type though.

**Review by michaelrj-google — APPROVED** [review]
LGTM

**Comment by nickdesaulniers** [comment]
Consider updating the commit message and PR description to note that this is only for fixedvector.

**Comment by jameshu15869** [comment]
Changed!

**Pull request merged**